main/src/main/scala/sbt/Defaults.scala

```diff
@@ -318,6 +318,7 @@ object Defaults extends BuildCommon {
       useLog4J :== SysProp.useLog4J,
       watchSources :== Nil, // Although this is deprecated, it can't be removed or it breaks += for legacy builds.
       skip :== false,
+      publishLocal / skip := (publish / skip).value, // So we don't break previous behaviour
```

Comment:
Could you add scripted test please? It should catch that the above line won't work in a subproject unless you move it to project setting.

Reply:
Should I place the scripted test in `sbt-app/src/sbt-test/project`?

Reply:
Indeed you were correct, this setting was in the wrong place (`globalSettings`) which meant that it didn't have the desired behaviour.

I added a scripted test in `project/skip` to verify that there is expected behaviour.